bouncer/views.py

```diff
@@ -25,11 +29,16 @@ def annotation(self):
                 id=self.request.matchdict["id"])
         except exceptions.NotFoundError:
             statsd.incr("views.annotation.404.annotation_not_found")
-            raise httpexceptions.HTTPNotFound()
+            raise httpexceptions.HTTPNotFound("Annotation not found")
 
         annotation_id = document["_id"]
         document_uri = document["_source"]["uri"]
 
+        if not (document_uri.startswith("http://") or
+                document_uri.startswith("https://")):
+            raise httpexceptions.HTTPUnprocessableEntity(
+                _("We can only link to http(s) URLs"))
```

Comment:
My suggestion here is: _"Sorry, currently we can only link to annotations of documents on the web. The annotated document must have an HTTP or HTTPS link."_ I'm trying to keep it user-friendly for non-experts (the first sentence, matches @robertknight's point 1) while also providing the necessary technical details for experts to fix the problem themselves when possible (the second sentence, matches Rob's point 2 I think), and also trying to keep it accurate (it specifically has to be an http(s) link), and concise.